src/core/EnvelopeAndLfoParameters.cpp

```diff
@@ -291,6 +291,8 @@ void EnvelopeAndLfoParameters::fillLevel( float * _buf, f_cnt_t _frame,
 						const f_cnt_t _release_begin,
 						const fpp_t _frames )
 {
+	m_paramMutex.lock();
```

Comment:
Please use a [QMutexLocker](https://doc.qt.io/qt-4.8/qmutexlocker.html) here to ensure safe unlocking in cases of exceptions, etc.